TACC/Core-Portal
# Task/fp 341  ui ux changes

## Overview: ##

UI/UX changes for PR #87.

## PR Status: ##


## Related Jira tickets: ##

* [FP-341](https://jira.tacc.utexas.edu/browse/FP-341)

## Summary of Changes: ##

### Relevant
- _Minor_: Prevent CSS lint error on `composes`.
- __New__: Include the `DataFilesSearchbar` component.
- __New__: Convert _some_ DataFiles CSS to CSS Modules.
- _Minor_: Add `className` support to `DataFilesBreadcrumbs`.
    - _So font size can be set via `DataFiles` and `DataFilesModal`._
- __Fix__: Improve _most_ CSS for DataFiles _(but __not__ table, which needs `InfiniteScrollTable`)_
    - _The layout __does__ change (to more closely match the design)._
- __New__: Add status message, dropdown filter placeholder, and "Back to All Files" link.
- __New__: Update `DataFilesSearchbar` tests.
    - _Replace `keypress`/`click` tests with single `submit` test._
    - _Add "has expected elements" test._
- _Minor_: Move `<div className="h-100">` to within `DataFilesTable`.\
    - _So that it is seen as paired with `Autosizer` Comment added._
- _Minor_: Update tests as necessary.
- _Minor_: Organize button styles and hierarchy of overrides.
- __New__: Add utility `taggedTemplates`.

## Testing Steps: ##
1. Load Data Files.
1. Enter search query.
1. Submit form via Enter key press(while search field is in focus).
    - _Ensure loading UI occurs._
    - _**Known Issue**: Results do __not__ reflect query because search is not integrated.¹_
1. Confirm that status message shows to the right.
    - _**Known Issue**: Result count is __not__ accurate because search is not integrated.¹_
1. Submit form via Button press.
    - _Ensure loading UI occurs._
    - _**Known Issue**: Results do __not__ reflect search because search is not integrated.¹_
1. Confirm that status message shows to the right.
    - _**Known Issue**: Result count is __not__ accurate because search is not integrated.¹_
    - _**Known Issue**: Status did __not__ clear, because search is not integrated.¹_
1. Click "Back to All Files".
1. Confirm that query and message are cleared.

### Footnotes
1. The branch `task/FP-341` did __not__ integrate search functionality. This is UI/UX branch. Some UX is hampered due to missing functionality. Comments have been added to the code for the affected UX.

## UI Photos:

- Before Search
    <img width="1269" alt="FP-341 UI Before Search" src="https://user-images.githubusercontent.com/62723358/88580657-90c3e480-d011-11ea-8d50-d0ba2276b251.png">
- After Search
    <img width="1269" alt="FP-341 UI After Search" src="https://user-images.githubusercontent.com/62723358/88580663-928da800-d011-11ea-949f-1245b2268b5b.png">

## Notes: ##

### Enhancements

1. Extra improvements to surrounding UI (within Data Files) were made.\
    _(Because doing properly-sized UI on the new elements, without fixing the old elements, would cause visual awfulness.)_

### Known Issues

1. Search is __not__ integrated with data listing.
1. The filter dropdown is __not__ real, because that comes in [FP-505](https://jira.tacc.utexas.edu/browse/FP-341).
